### PR TITLE
Add zoom reset function

### LINKED ISSUE #287: Reset not implemented
When resetting a Moodle course that has a Zoom activity the page shows a message telling that this activity hasn't implemented that feature.

### PR DESCRIPTION
**Pull request by smbader**
- Reset student grades
- Reset user data tables
- Reset form definitions and language strings

TODO: Behat tests.

Fixes #287

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -758,6 +758,87 @@ function zoom_update_grades(stdClass $zoom, $userid = 0) {
...
+
+    $DB->delete_records_select('zoom_meeting_recordings_view',
+        'recordingsid IN (SELECT zmr.id
+                         FROM {zoom_meeting_recordings} zmr
+                         JOIN {zoom} z ON z.id = zmr.zoomid
+                        WHERE z.course = ?)', array($data->courseid));
+    $status[] = array(
+        'component' => $componentstr,
+        'item' => get_string('meetingrecordingsviewdeleted', 'zoom'),
+        'error' => false);
```
Comment (jrchamp) [inline lib.php:816]:
```suggestion
        'error' => false,
    );
```
```suggestion
        'error' => false);
```

Comment (jrchamp) [inline lib.php:816]:
```suggestion
        'item' => get_string('meetingrecordingviewsdeleted', 'zoom'),
        'error' => false,
    );
```

Comment (jrchamp) [inline lib.php:816]:
```suggestion
        'item' => get_string('meetingrecordingviewsdeleted', 'zoom'),
        'error' => false,
    );
```
```suggestion
        'item' => get_string('meetingrecordingsviewdeleted', 'zoom'),
        'error' => false);
```

**Review by jrchamp — COMMENTED** [review]
Looks good! Only a few small adjustments. Then once testing confirms it's working, this should be an easy merge. 🤞

**Comment by sgrandh3** [comment]
Was able to verify that the student grades and language strings are deleted after the reset. During reset , the instructor can check a box to delete the user data in the zoom activity. All are cleared as expected.

**Review by jrchamp — APPROVED** [review]
The code makes sense and testing was a success, so now we'll have reset functionality available!

**Pull request merged**